packages/calling/src/CallingClient/CallingClient.test.ts

```diff
@@ -234,254 +162,401 @@ describe('CallingClient Tests', () => {
-      expect(infoSpy).toHaveBeenCalledWith('Updating region and country from the SDK config', {
-        file: 'CallingClient',
-        method: 'getMobiusServers',
+    it('Simulate a network flap before initial registration is done', async () => {
+      expect(line.getRegistrationStatus()).toEqual(MobiusStatus.ACTIVE);
+
+      reg.deregister();
+      reg.setActiveMobiusUrl(undefined);
+
+      jest.clearAllMocks();
+
+      /* Set mercury connection to be down and execute a delay of 2.5 seconds */
+      webex.internal.mercury.connected = false;
+      jest.advanceTimersByTime(NETWORK_FLAP_TIMEOUT + 500);
+
+      /* We should be detecting the network flap */
+      expect(warnSpy).toBeCalledWith(
+        'Network has flapped, waiting for mercury connection to be up',
+        {file: CALLING_CLIENT_FILE, method: NETWORK_CHANGE_DETECTION_UTIL}
+      );
+
+      /* Set mercury connection to be up and execute a delay of 2.5 seconds */
+      webex.internal.mercury.connected = true;
+      jest.advanceTimersByTime(NETWORK_FLAP_TIMEOUT + 500);
+
+      await flushPromises();
+
+      /* We should be detecting the network recovery */
+      expect(logSpy).toBeCalledWith('Mercury connection is up again, Re-registering with Mobius', {
+        file: REGISTRATION_FILE,
+        method: 'handleConnectionRestoration',
       });
+
+      /*
+       * When initial registration is not done, network flap
+       * will not trigger de-registration/registration
+       */
+      expect(webex.request).not.toBeCalled();
+      expect(restoreSpy).not.toBeCalled();
+      expect(registerSpy).not.toBeCalled();
+    });
+
+    it('Simulate a network flap with 1 active call', async () => {
+      expect(line.getRegistrationStatus()).toEqual(MobiusStatus.ACTIVE);
+
+      /** create a new call */
+      reg.callManager.createCall();
+      expect(Object.keys(reg.callManager.getActiveCalls()).length).toBe(1);
+
+      /* Set mercury connection to be down and execute a delay of 2.5 seconds */
+      webex.internal.mercury.connected = false;
+      jest.advanceTimersByTime(NETWORK_FLAP_TIMEOUT + 500);
+      await flushPromises();
+
+      /* We should be detecting the network flap */
+      expect(warnSpy).not.toBeCalledWith(
+        'Network has flapped, waiting for mercury connection to be up',
+        {file: CALLING_CLIENT_FILE, method: 'handleConnectionRestoration'}
+      );
+
+      /* Set mercury connection to be up and execute a delay of 2.5 seconds */
+      webex.internal.mercury.connected = true;
+      jest.advanceTimersByTime(NETWORK_FLAP_TIMEOUT + 500);
+
+      await flushPromises();
+
+      /* We should be detecting the network recovery */
+      expect(logSpy).not.toBeCalledWith(
+        'Mercury connection is up again, Re-registering with Mobius',
+        {file: REGISTRATION_FILE, method: 'handleConnectionRestoration'}
+      );
+
+      expect(registerSpy).not.toBeCalledWith(true);
+      expect(registerSpy).toBeCalledTimes(0);
```

Comment:
first expect statement should suffice, we don't need toBeCalledTimes(0)

Reply:
Added by mistake while creating this new PR. Corrected it.